lib/Optimizer/GraphOptimizer/NodeSplitting.cpp

```diff
@@ -0,0 +1,1560 @@
+  outputIdxMask[splitOutputIdx] = true;
+  for (const auto &outputIdxMap : outputIdxAndMaps) {
+    RETURN_ERR_IF_NOT(outputIdxMap.first < node->getNumResults(),
+                      "Invalid output index for output range map!");
+    outputIdxMask[outputIdxMap.first] = true;
+  }
+  RETURN_ERR_IF_NOT(
+      std::find(outputIdxMask.begin(), outputIdxMask.end(), false) ==
+          outputIdxMask.end(),
+      "Not all output indices and maps were provided for splitting node!");
+
+  // Get split output range.
+  SliceRange splitOutputRange = SliceRange(node->getType(splitOutputIdx));
+
+  // Verify the input slice range maps.
+  for (const auto &inputIdxMap : inputIdxAndMaps) {
+    SliceRange inputRange =
+        SliceRange(node->getNthInput(inputIdxMap.first).getType());
+    if (std::find(weakOutToInMappingNodeKinds.begin(),
+                  weakOutToInMappingNodeKinds.end(),
+                  node->getKind()) != weakOutToInMappingNodeKinds.end()) {
+      // Verify weak mapping.
+      RETURN_ERR_IF_NOT(isMappingIncluded(inputIdxMap.second,
+                                          {splitOutputRange}, {inputRange}),
+                        "Invalid input range map for splitting node!");
+    } else {
+      // Verify exact mapping.
+      RETURN_ERR_IF_NOT(
+          isMappingExact(inputIdxMap.second, {splitOutputRange}, {inputRange}),
+          "Invalid input range map for splitting node!");
+    }
+  }
+
+  // Verify the output slice range maps.
+  for (const auto &outputIdxMap : outputIdxAndMaps) {
+    SliceRange outputRange = SliceRange(node->getType(outputIdxMap.first));
+    RETURN_ERR_IF_NOT(
+        isMappingExact(outputIdxMap.second, {splitOutputRange}, {outputRange}),
+        "Invalid output range map for splitting node!");
+  }
+
+  return Error::success();
+}
+
+///===---------------------------------------------------------------------===//
+///                            verifySplitNodes
+///===---------------------------------------------------------------------===//
+/// Function to verify the split nodes.
+static Expected<bool>
+verifySplitNodes(const Node *node, dim_t splitOutputIdx,
+                 const llvm::ArrayRef<SliceRange> &splitOutputSlices,
+                 const llvm::ArrayRef<OpIdxAndMap> &inputIdxAndMaps,
+                 const llvm::ArrayRef<OpIdxAndMap> &outputIdxAndMaps,
+                 const SplitNodeConstraint *splitConstraint,
+                 const SplitNodeModifier &splitNodeModifier) {
+
+  // Create temporary nodes to make verifications without adding them to
+  // the graph in order to avoid the pollution of the graph with nodes
+  // which could be invalid or could not meet all the constraints and
+  // hence be later removed from the graph.
+  bool splitNodesCheck = true;
+  std::vector<Node *> splitNodes;
+  std::list<std::unique_ptr<SliceNode>> inputSliceNodes;
+  std::list<Type> inputTypes;
+  std::list<Type> outputTypes;
```

Comment:
It is about the lifetime of the objects. They must be alive at the time the split context is passed to the constraint for verification.
The reason I have this entire function using orphan nodes/types is because for some usecases (search by memory constraint) I could go through this function thousands of times and most of the times I would need to discard the inserted nodes/types with 0 reutilization.
Therefore everything that happens inside the function `verifySplitNodes` I want to happen in `orphan mode` without needing to spend extra time later for the cleaning up the Function/Module.